src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -262,14 +267,7 @@ internal Silo(SiloInitializationParameters initializationParams)
                     services.TryAddSingleton<ClientObserversPlacementDirector>();
                     
                     services.TryAddSingleton<Func<IGrainRuntime>>(sp => () => sp.GetRequiredService<IGrainRuntime>());
-                    if (usingCustomServiceProvider)
-                    {
-                        services.TryAddSingleton<GrainCreator>();
-                    }
-                    else
-                    {
-                        services.TryAddSingleton(sp => new GrainCreator(null, sp.GetRequiredService<Func<IGrainRuntime>>()));
-                    }
+                    services.TryAddSingleton<GrainCreator>();
```

Comment:
I'm trying to understand if this is correct. Today we explicitly pass `null` as a first argument to `new GrainCreator`, which triggers it to use `Activator.CreateInstance(type)` for instantiating grain activations. This will make it always use `ActivatorUtilities.CreateFactory(type, Type.EmptyTypes)`.

Why do we need to change this? Seems unrelated to the stated purpose of the PR.